Test suite failures

The Makefile cannot build in a standard go workspace. It fails as follows:
```
$ make test_verm
mkdir -p src/github.com/willbryant
( cd src/github.com/willbryant && ln -sf ../../../ verm)
ln: verm/..: File exists
make: *** [verm] Error 1
Exit 2
```

And when verm is built manually the test suite fails:
```
$ go build && make test_verm
BUNDLE_GEMFILE=test/Gemfile bundle exec testrb test/get_files_test.rb test/create_files_multipart_test.rb test/create_files_raw_test.rb test/health_check_test.rb test/replication_put_test.rb test/replication_missing_test.rb test/replication_propagation_test.rb test/replication_topology_test.rb test/read_forwarding_test.rb
Run options: 

# Running tests:

[68/79] ReplicationPropagationTest#test_retries_propagation_if_slave_unavailable = 0.25 s                                      
  1) Failure:
test_retries_propagation_if_slave_unavailable(ReplicationPropagationTest) [/Users/james/Development/go/src/github.com/willbryant/verm/test/replication_propagation_test.rb:143]:
replication error was not logged

Finished tests in 19.738436s, 4.0023 tests/s, 39.8208 assertions/s.                                                      
79 tests, 786 assertions, 1 failures, 0 errors, 0 skips

ruby -v: ruby 2.1.7p400 (2015-08-18 revision 51632) [x86_64-darwin15.0]
make: *** [test_verm] Error 1
Exit 2
```

Here's my environment.
```
$ go version
go version go1.7.4 darwin/amd64
```
```
$ go env
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/james/Development/go"
GORACE=""
GOROOT="/usr/local/Cellar/go/1.7.4/libexec"
GOTOOLDIR="/usr/local/Cellar/go/1.7.4/libexec/pkg/tool/darwin_amd64"
CC="clang"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/h2/_dbq0vpd7lz9yr3ntqgx11mr0000gn/T/go-build041211897=/tmp/go-build -gno-record-gcc-switches -fno-common"
CXX="clang++"
CGO_ENABLED="1"
```
```
$ ruby --version
ruby 2.1.7p400 (2015-08-18 revision 51632) [x86_64-darwin15.0]
```

I also had trouble using `testrb`. It is either not included in newer versions of ruby or you need to specify the dependency in Gemfile.

--- a/Makefile
+++ b/Makefile
@@ -17,4 +17,4 @@ clean:
 	rm -rf verm src
 
 test_verm: verm
-	BUNDLE_GEMFILE=test/Gemfile bundle exec testrb test/get_files_test.rb test/create_files_multipart_test.rb test/create_files_raw_test.rb test/health_check_test.rb test/replication_put_test.rb test/replication_missing_test.rb test/replication_propagation_test.rb test/replication_topology_test.rb test/read_forwarding_test.rb
+	BUNDLE_GEMFILE=test/Gemfile bundle exec rake -f test/Rakefile test

--- a/test/Gemfile
+++ b/test/Gemfile
@@ -1,4 +1,5 @@
 source 'https://rubygems.org'
 
-gem "minitest", "~> 4.0"
+gem "minitest", "~> 5.10"
 gem "byebug"
+gem "rake"

--- a/test/Rakefile
+++ b/test/Rakefile
@@ -1,0 +1,5 @@
+require 'rake/testtask'
+
+Rake::TestTask.new do |t|
+  t.pattern = "test/*_test.rb"
+end

--- a/test/create_files_raw_test.rb
+++ b/test/create_files_raw_test.rb
@@ -24,7 +24,7 @@ class CreateFilesRawTest < Verm::TestCase
 
   def test_saves_same_gzipped_file_compressed_differently_to_same_path
     recompressed = gzip(fixture_file_data('simple_text_file'))
-    assert_not_equal recompressed, fixture_file_data('simple_text_file.gz')
+    refute_equal recompressed, fixture_file_data('simple_text_file.gz')
 
     location1 =
       post_file :path => '/foo',

--- a/test/create_files_tests.rb
+++ b/test/create_files_tests.rb
@@ -111,7 +111,7 @@ module CreateFilesSharedTests
                   :type => 'application/octet-stream',
                   :expected_extension => nil)
       
-      assert_not_equal first_file_location, different_file_location
+      refute_equal first_file_location, different_file_location
     end
   end
 

--- a/test/get_files_test.rb
+++ b/test/get_files_test.rb
@@ -74,7 +74,7 @@ class GetFilesTest < Verm::TestCase
     assert_statistics_change(:get_requests => 2) do
       copy_arbitrary_file_to('somefiles', nil)
       response = get :path => @location
-      assert_not_nil response['etag']
+      refute_nil response['etag']
       
       get :path => @location,
           :headers => {'if-none-match' => response['etag']},

--- a/test/net_http_multipart_post.rb
+++ b/test/net_http_multipart_post.rb
@@ -9,6 +9,7 @@ module Net
       def initialize(*args)
         super(*args)
 
+        @added_final_boundary = nil
         chrs = "0123456789abcdefghijklmnopqrstuvwxyz"
         random_str = (0..11).collect { chrs[rand(36)] }.join
         @boundary = ".multipart_boundary_#{random_str}.".freeze

--- a/test/replication_missing_test.rb
+++ b/test/replication_missing_test.rb
@@ -11,7 +11,7 @@ class ReplicationMissingTest < Verm::TestCase
   end
 
   def assert_ungzipped_body(content, response)
-    assert_equal nil, response['content-encoding']
+    assert_nil response['content-encoding']
     assert_equal content, response.body
   end
 

--- a/test/replication_propagation_test.rb
+++ b/test/replication_propagation_test.rb
@@ -140,9 +140,9 @@ class ReplicationPropagationTest < Verm::TestCase
     }, changes)
 
     unless ENV['VALGRIND'] || ENV['NO_CAPTURE_STDERR'].to_i > 0
-      assert_match /#{VERM_SPAWNER.port}:( getsockopt:)? connection refused/,
+      assert_match(/#{VERM_SPAWNER.port}:( getsockopt:)? connection refused/,
         File.read(REPLICATION_MASTER_VERM_SPAWNER.capture_stderr_in).downcase,
-        "replication error was not logged"
+        "replication error was not logged")
     end
 
     VERM_SPAWNER.start_verm

--- a/test/test_helper.rb
+++ b/test/test_helper.rb
@@ -1,5 +1,4 @@
-require 'rubygems'
-require 'test/unit'
+require 'minitest/autorun'
 require 'fileutils'
 require 'byebug'
 require File.expand_path(File.join(File.dirname(__FILE__), 'net_http_multipart_post'))
@@ -15,7 +14,7 @@ VERM_SPAWNER = VermSpawner.new(verm_binary, verm_data, :mime_types_file => mime_
 REPLICATION_MASTER_VERM_SPAWNER = VermSpawner.new(verm_binary, "#{verm_data}_replica", :mime_types_file => mime_types_file, :port => VERM_SPAWNER.port + 1, :replicate_to => VERM_SPAWNER.host, :capture_stdout_in => captured_stdout_filename, :capture_stderr_in => captured_stderr_filename)
 
 module Verm
-  class TestCase < Test::Unit::TestCase
+  class TestCase < Minitest::Test
     undef_method :default_test if instance_methods.include? 'default_test' or
                                   instance_methods.include? :default_test
 
@@ -24,6 +23,7 @@ module Verm
     end
   
     def setup
+      @multipart = nil
       VERM_SPAWNER.setup(extra_spawner_options)
     end
 
@@ -103,12 +103,21 @@ module Verm
       end
 
       assert_equal options[:expected_response_code] || 200, response.code.to_i, "The response didn't have the expected code"
-      assert_equal options[:expected_content_type], response.content_type, "The response had an incorrect content-type" if options.has_key?(:expected_content_type)
-      assert_equal options[:expected_content_length], response.content_length, "The response had an incorrect content-length" if options.has_key?(:expected_content_length)
-      assert_equal options[:expected_content_encoding], response['content-encoding'], "The response had an incorrect content-encoding" if options.has_key?(:expected_content_encoding)
-      assert_equal options[:expected_content], response.body, "The response had incorrect content" if options.has_key?(:expected_content)
+      assert_equal_or_nil options[:expected_content_type], response.content_type, "The response had an incorrect content-type" if options.has_key?(:expected_content_type)
+      assert_equal_or_nil options[:expected_content_length], response.content_length, "The response had an incorrect content-length" if options.has_key?(:expected_content_length)
+      assert_equal_or_nil options[:expected_content_encoding], response['content-encoding'], "The response had an incorrect content-encoding" if options.has_key?(:expected_content_encoding)
+      assert_equal_or_nil options[:expected_content], response.body, "The response had incorrect content" if options.has_key?(:expected_content)
       
       response
+    end
+
+    # Prevent "Use assert_nil if expecting nil... This will fail in MT6" warnings
+    def assert_equal_or_nil(expected, actual, message=nil)
+      if expected.nil?
+        assert_nil actual, message
+      else
+        assert_equal expected, actual, message
+      end
     end
 
     def expected_filename(location, options = {})
@@ -219,7 +228,7 @@ module Verm
     end
 
     def calculate_statistics_change(before, after)
-      results = after.inject({}) {|results, (k, v)| results[k] = v - before[k] unless v == before[k]; results}
+      after.inject({}) {|results, (k, v)| results[k] = v - before[k] unless v == before[k]; results}
     end
 
     def assert_statistics_change(expected_change, options = {})

--- a/test/verm_spawner.rb
+++ b/test/verm_spawner.rb
@@ -76,7 +76,7 @@ class VermSpawner
         puts e
         exit 1
       end
-      exec *exec_args
+      exec(*exec_args)
     end
   end
   

